src/include/duckdb.h

```diff
@@ -916,6 +916,12 @@ Clear the params bind to the prepared statement.
 */
 DUCKDB_API duckdb_state duckdb_clear_bindings(duckdb_prepared_statement prepared_statement);
 
+/*!
+Retrieve the index of the parameter for the prepared statement, identified by name
+*/
+DUCKDB_API duckdb_state duckdb_bind_parameter_index(duckdb_prepared_statement prepared_statement, idx_t *param_idx_out,
```

Comment:
I think it would be valuable to be able to list the set of parameters as well, for example we could add a `const char *duckdb_parameter_name(duckdb_prepared_statement prepared_statement, idx_t index)` function that can be used together with the `duckdb_nparams` function

Reply:
Added it, still needs testing though